payjoin/src/receiver/mod.rs

```diff
@@ -233,26 +237,22 @@ impl MaybeInputsSeen {
     /// proposes a PayJoin PSBT as a new Original PSBT for a new PayJoin.
     pub fn check_no_inputs_seen_before(
         self,
-        is_known: impl Fn(&Script) -> bool,
-    ) -> Result<OutputsUnknown, RequestError> {
-        let mut known_script = None;
-        self.psbt.input_pairs().for_each(|input| {
-            let script = &input
-                .previous_txout()
-                // This error should already be handled by the first check
-                .unwrap_or_else(|e: crate::psbt::PrevTxOutError| {
-                    panic!("Unexpected error: {:?}", e)
-                })
-                // Once this closure returns a Result, we can use this instead:
-                //.map_err(|e| RequestError::from(InternalRequestError::PrevTxOut(e)))?.script_pubkey;
-                .script_pubkey;
-            if is_known(script) {
-                known_script = Some(script.clone());
-            }
-        });
-        if let Some(script) = known_script {
-            return Err(RequestError::from(InternalRequestError::InputSeen(script)));
-        }
+        is_known: impl Fn(&Script) -> Result<bool, Error>,
+    ) -> Result<OutputsUnknown, Error> {
+        self.psbt
+            .input_pairs().try_for_each(|input| {
+                if let Ok(txo) = input.previous_txout() {
```

Comment:
See here's another of these discrepancies. The error is discarded here because it would have already been dealth with on psbt parsing.